mlflow/llama_index/pyfunc_wrapper.py

```diff
@@ -43,9 +43,8 @@ def __init__(
     ):
         self.index = index
         self.model_config = model_config or {}
-        self.predict_callable = self._build_engine_method()
 
-    def _build_engine_method(self) -> Callable:
+    def _predict_single(self, *args, **kwargs) -> Any:
```

Comment:
What are you thoughts on externalizing the ABC to its own base module within a pyfunc directory in the flavor, as well as migrating each distinct implementation to its own file? 
I'm wondering how many engine types this library will eventually have...

Reply:
Yea I think it's good idea. Lemme file a follow-up PR for it.